### ISSUE TITLE
"make install" installs development packages as well

### ISSUE THREAD

**Issue by andy-maier** [zhmcclient/zhmc-ansible-modules#462]
Initial investigation:

The reason that 'make install' installs the development dependencies as well, is that the ansible-doc-extractor tool is needed to build the distribution archive needed for the install, and that comes with the development environment.

It would be possible to install specifically the ansible-doc-extractor package for install, but that would require using the dev-requirements.txt file as a constraints file for that install. Doing so requires that pip supports the git+http URLs we use in the dev-requirements.txt as constraints. Support for named URL constraints was added in pip 21.1 (See discussion in https://github.com/pypa/pip/issues/8210 and https://github.com/pypa/pip/issues/8253). However, pip 21.1 no longer supports Python 2.7 and 3.5.

Bottom line: This issue can only be resolved once we drop Python 2.7 and 3.5 support which allows us to increase the minimum version of pip to >=21.1.

**Comment by andy-maier** [comment]
Another possible solution is to no longer include the documentation .rst files in the collection package, then they do not need to be generated in the first place for the package build.
